src/SprykerSdk/Zed/ComposerConstrainer/Business/Validator/StrictConstraintValidator.php

```diff
@@ -0,0 +1,288 @@
+            ->setIsDev($isDev);
+    }
+
+    /**
+     * Specification:
+     * - Calculates what version is defined in composer.lock for a given module.
+     *
+     * @param \Generated\Shared\Transfer\ComposerConstraintModuleInfoTransfer $usedModuleInfo
+     * @param \Generated\Shared\Transfer\ComposerConstraintTransfer|null $lockConstraint
+     *
+     * @return \Generated\Shared\Transfer\ComposerConstraintModuleInfoTransfer
+     */
+    protected function setLockModuleInfo(
+        ComposerConstraintModuleInfoTransfer $usedModuleInfo,
+        ?ComposerConstraintTransfer $lockConstraint = null
+    ): ComposerConstraintModuleInfoTransfer {
+        if ($lockConstraint === null) {
+            return $usedModuleInfo;
+        }
+
+        return $usedModuleInfo->setLockedVersion($lockConstraint->getVersion());
+    }
+
+    /**
+     * Specification
+     * - Used modules have either "public API" (^) or customized (~) lock
+     * - Defined constraint lock is either missing or major (^), minor (~) or patch ("")
+     * - If defined constraint is missing then module usage constraint drives the expectation
+     * - If defined constraint is present then the smaller lock is the expectation
+     *
+     * @param \Generated\Shared\Transfer\ComposerConstraintModuleInfoTransfer $moduleInfo
+     *
+     * @return \Generated\Shared\Transfer\ComposerConstraintModuleInfoTransfer
+     */
+    protected function setExpectation(ComposerConstraintModuleInfoTransfer $usedModuleInfo): ComposerConstraintModuleInfoTransfer
+    {
+        $moduleUsageConstraintLock = $usedModuleInfo->getIsCustomized() ? '~' : '^';
+        $definedConstraintLock = $usedModuleInfo->getDefinedConstraintLock();
+        $hasDefinedVersion = !empty($usedModuleInfo->getDefinedVersion());
+
+        $expectedConstraintLock = $moduleUsageConstraintLock;
+        if ($hasDefinedVersion) {
+            $expectedConstraintLock = $definedConstraintLock;
+            if ($definedConstraintLock === '^' && $moduleUsageConstraintLock === '~') {
+                $expectedConstraintLock = '~';
+            }
+        }
+
+        return $usedModuleInfo
+            ->setExpectedConstraintLock($expectedConstraintLock)
+            ->setExpectedVersion($usedModuleInfo->getDefinedVersion() ?: $usedModuleInfo->getLockedVersion());
+    }
+
+    /**
+     * Specification
+     * - Ignored packages are removed.
+     * - Line count MUST be zero to be considered properly developed.
+     * - Expected and defined constraint lock need to match to be considered will configured.
+     *
+     * @param \Generated\Shared\Transfer\ComposerConstraintTransfer[] $composerConstraintTransfers
+     *
+     * @return \Generated\Shared\Transfer\ComposerConstraintTransfer[]
+     */
+    protected function removeCorrectPackages(array $composerConstraintTransfers): array
+    {
+        $ignoredPackages = '#(' . implode('|', $this->config->getStrictValidationIgnoredPackages()) . ')#';
```

Comment:
Given the package names, I think you need ^ and $ here to make sure not to catch false positives in the long run
e.g. spryker/kernel-helper or alike in the far future (we never know).

Reply:
i want to support magic here. On project, we actually group-ignore packages like this. This is why in Config i had the `^..$` around packages names and you asked me to remove them ;)

Reply:
OK
Well, to remove them inside the config array, not inside the actual regex, was my thinking :)